server/odc-service/src/test/java/com/oceanbase/odc/service/sqlcheck/MySQLCheckerTest.java

```diff
@@ -1362,6 +1362,212 @@ public void check_wrongPlExists_violationGenerated() {
+                "--------------------------------------------------",
+                "|0 |DISTRIBUTED INSERT|    |0       |20          |",
+                "|1 |└─EXPRESSION      |    |3       |1           |",
+                "==================================================");
+
+        JdbcTemplate jdbcTemplate = Mockito.mock(JdbcTemplate.class);
+
+        Mockito.when(jdbcTemplate.query(Mockito.anyString(), Mockito.any(RowMapper.class)))
+                .thenReturn(resultSet);
+        DefaultSqlChecker deleteChecker = new DefaultSqlChecker(DialectType.OB_MYSQL, "$$",
+                Collections.singletonList(
+                        new MySQLAffectedRows(2L, DialectType.OB_MYSQL, jdbcTemplate)));
+        List<CheckViolation> actualDelete = deleteChecker.check(delete);
+        Assert.assertEquals(1, actualDelete.size());
+    }
+
+    @Test
+    public void check_restrictSqlAffectedRows4OBOthers() {
+        String select =
+                "select id, name, age from users where id in ('1', '2')";
+        List<String> resultSet = Arrays.asList(
+                "==================================================",
+                "|ID|OPERATOR          |NAME|EST.ROWS|EST.TIME(us)|",
+                "--------------------------------------------------",
+                "|0 |DISTRIBUTED INSERT|    |4       |20          |",
+                "|1 |└─EXPRESSION      |    |4       |1           |",
+                "==================================================");
+        JdbcTemplate jdbcTemplate = Mockito.mock(JdbcTemplate.class);
+
+        Mockito.when(jdbcTemplate.query(Mockito.anyString(), Mockito.any(RowMapper.class)))
+                .thenReturn(resultSet);
+        DefaultSqlChecker selectChecker = new DefaultSqlChecker(DialectType.OB_MYSQL, "$$",
+                Collections.singletonList(
+                        new MySQLAffectedRows(2L, DialectType.OB_MYSQL, jdbcTemplate)));
+        List<CheckViolation> actualSelect = selectChecker.check(select);
+        Assert.assertEquals(0, actualSelect.size());
+    }
+
+    @Test
+    public void check_restrictSqlAffectedRows4OBError() {
+        String update =
+                "update users set name = 'a1-bot' where id = '1'";
+
+        JdbcTemplate jdbcTemplate = Mockito.mock(JdbcTemplate.class);
+
+        Mockito.when(jdbcTemplate.query(Mockito.anyString(), Mockito.any(RowMapper.class)))
+                .thenThrow(new RuntimeException("Failed to execute sql"));
+        try {
+            DefaultSqlChecker errorChecker = new DefaultSqlChecker(DialectType.OB_MYSQL, "$$",
+                    Collections.singletonList(
+                            new MySQLAffectedRows(2L, DialectType.OB_MYSQL, jdbcTemplate)));
+            List<CheckViolation> actualError = errorChecker.check(update);
+        } catch (RuntimeException e) {
+            Assert.assertEquals("Failed to execute sql: " + "EXPLAIN " + update,
```

Comment:
if no exception thrown, the Assert will not be execute, so please avoid try catch in unit testing logic.
use @Rule ExpectedExcetion mechanism instead.

Reply:
got it, it's done.